security/pkg/nodeagent/cache/secretcache.go

```diff
@@ -304,6 +305,24 @@ func (sc *SecretCache) rotate() {
 }
 
 func (sc *SecretCache) generateSecret(ctx context.Context, token, resourceName string, t time.Time) (*model.SecretItem, error) {
+	// If node agent works as ingress gateway agent, searches for kubernetes secret instead of sending
+	// CSR to CA.
+	if sc.fetcher.UseCaClient == false {
+		secretItem, exist := sc.fetcher.FindIngressGatewaySecret(resourceName)
+		if !exist {
+			return nil, fmt.Errorf("cannot find secret %s for ingress gateway", resourceName)
+		}
+
+		return &model.SecretItem{
+			CertificateChain: secretItem.CertificateChain,
```

Comment:
customer bring cert is only normal key/cert pair, or also include customer root cert ? if also include root cert, more logic is needed

Reply:
Good question. Per offline discussion with @myidpt , customer does not add root cert. Only  key/cert pairs are used for ingress gateway.